<launch>
  <arg name="use_robot_name" default="papras"/>

  <!-- These are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="gui" default="true"/>
  <arg name="paused" default="true"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="with_gripper" default="false"/>
  <arg name="use_platform"  default="false"/>
  <arg name="synced_gui"  default="false"/>

  <!-- Load the URDF into the ROS Parameter Server -->
  <include file="$(find open_manipulator_p_description)/launch/open_manipulator_p_upload.launch">
    <arg name="with_gripper" value="$(arg with_gripper)"/>
  </include>

  <!-- Launch the Control GUI -->
  <group if="$(arg synced_gui)">
    <include file="$(find open_manipulator_p_control_gui_synced)/launch/open_manipulator_p_control_gui_synced.launch">
        <arg name="with_gripper" value="$(arg with_gripper)"/>
    </include>
  </group>

  <!-- Launch Rviz -->
  <include file="$(find open_manipulator_p_description)/launch/open_manipulator_p_rviz.launch">
      <arg name="with_gripper" value="$(arg with_gripper)"/>
  </include>

  <!-- BEGIN ROBOT 1-->
  <group ns="robot1">
    <param name="tf_prefix" value="robot1_tf" />

    <include file="$(find papras_gazebo)/launch/one_arm.launch" >
      <arg name="init_pose" value="-x -0.3 -y 0.1 -z 0.9 -R 0.0 -P 0.0 -Y 1.57" />
      <arg name="robot_name"  value="robot1" />
      <arg name="with_gripper" value="$(arg with_gripper)"/>
      <arg name="usb_port" value="/dev/ttyUSB0"/>
      <arg name="arm_id" value="1"/>
      <arg name="use_platform" value="$(arg use_platform)"/>
      <arg name="synced_gui" value="$(arg synced_gui)"/>
    </include>
  </group>

  <!-- BEGIN ROBOT 2-->
  <group ns="robot2">
    <param name="tf_prefix" value="robot2_tf" />
    <include file="$(find papras_gazebo)/launch/one_arm.launch" >
      <arg name="init_pose" value="-x -0.3 -y -0.5 -z 0.9 -R 0.0 -P 0.0 -Y 1.57" />
      <arg name="robot_name"  value="robot2" />
      <arg name="with_gripper" value="$(arg with_gripper)"/>
      <arg name="usb_port" value="/dev/ttyUSB1"/>
      <arg name="arm_id" value="2"/>
      <arg name="use_platform" value="$(arg use_platform)"/>
      <arg name="synced_gui" value="$(arg synced_gui)"/>
    </include>
  </group>

  <!-- BEGIN ROBOT 3-->
  <group ns="robot3">
    <param name="tf_prefix" value="robot3_tf" />
    <include file="$(find papras_gazebo)/launch/one_arm.launch" >
      <arg name="init_pose" value="-x 0.3 -y 0.0 -z 0.9 -R 0.0 -P 0.0 -Y -1.57" />
      <arg name="robot_name"  value="robot3" />
      <arg name="with_gripper" value="$(arg with_gripper)"/>
      <arg name="usb_port" value="/dev/ttyUSB2"/>
      <arg name="arm_id" value="3"/>
      <arg name="use_platform" value="$(arg use_platform)"/>
      <arg name="synced_gui" value="$(arg synced_gui)"/>
    </include>
  </group>

</launch>
